### PR TITLE
Switch documentation search to use Algolia

### PR DESCRIPTION
**Pull request by bandish-shah**
## Description of changes:

This PR switches the default documentation search engine to be Algolia.  Along for the ride it also fixes some indentation warnings in docstrings.

## Issue #, if available:

N/A

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [x] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [ ] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [ ] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#running-tests))
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: streaming/base/dataset.py [hunk 1]
```diff
@@ -98,30 +98,29 @@ class StreamingDataset(IterableDataset):
...
-          * Partitioning:
-              * ``partition_algo``
-              * ``num_canonical_nodes``
-              * ``batch_size``
-          * Shuffling:
-              * ``shuffle``
-              * ``shuffle_algo``
-              * ``shuffle_seed``
-              * ``shuffle_block_size``
+        * One or more Streams (you must provide either ``streams`` or ``remote``/``local``):
```
Comment (knighton) [inline streaming/base/dataset.py:101]: what's the scheme of this indentation change?

Comment (karan6181) [reply streaming/base/dataset.py:101]: I think either it has to be 4 spaces or 2 spaces but not mixed. Previously, it was mixed. Here it is 4 space

File: streaming/base/stream.py [hunk 2]
```diff
@@ -29,24 +29,24 @@ class Stream:
     Stream init takes three kinds of arguments:
       * At least one of ``remote`` and ``local`` must exist. If no ``remote``, the data must be
         local. If no ``local``, we cache to a temp directory.
-          * ``remote``
-          * ``local``
+        * ``remote``
```
[the remark this answers is not in the document]
Comment (karan6181) [reply streaming/base/stream.py:32]: Here it is 2 space. Can we make it uniform to either 2 space or 4 ?

File: streaming/base/dataset.py (hunk 1 above)
Comment (bandish-shah) [reply streaming/base/dataset.py:101]: Need to play with this some more, Sphinx was throwing a warning locally so I was messing around with it.

Comment (knighton) [reply streaming/base/dataset.py:101]: was doing 4-space, not counting the preceding "* "

Comment (bandish-shah) [reply streaming/base/dataset.py]: Ok think I fixed it, the initial asterisk of the main bullet needs to be aligned with the text body above (no indentation).  Subsequent nested sub bullets need to be separated by a blank line and indented by two spaces.  Gotta love RST

File: streaming/base/stream.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bandish-shah) [reply streaming/base/stream.py]: Resolved see above.

**Review by knighton — APPROVED** [review]

**Pull request merged**